### PR TITLE
Added: Time-derivative of time-dependent spatial functions

### PR DESCRIPTION
**Pull request by kmokstad**
Needed to support arclength control in nonlinear elasticity simulation with (pseudo)time-dependent loads or neumann conditions

### CODE REVIEW

File: src/Utility/Functions.C [hunk 1]
```diff
@@ -464,6 +480,26 @@ Real Interpolate1D::evaluate (const Vec3& X) const
 }
 
 
+Real Interpolate1D::deriv (const Vec3& X, int ddir) const
+{
+  Real res = Real(0);
+  if (ddir = dir+1)
```
Comment (akva2) [inline src/Utility/Functions.C:486]: == ?

Comment (kmokstad) [reply src/Utility/Functions.C]: Indeed. Fixed.

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=142 ifem-finitedeformation=35 ifem-openfrac=76 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=142 ifem-finitedeformation=35 ifem-openfrac=76 please

**Pull request merged**